File: 2021_Dev-Matching-the_first_half/The_highest_&_lowest_ranks/solution.js
function solution(lottos, win_nums) {
  let answer = [];
  let rank = [6, 6, 5, 4, 3, 2, 1];
  let max = 0;
  let min = lottos.filter((lotto) => {
    if (lotto === 0) max++;
    return win_nums.includes(lotto);
  }).length;

  answer[0] = rank[max + min];
  answer[1] = rank[min];

  return answer;
}

File: 2021_Dev-Matching-the_first_half/The_highest_&_lowest_ranks/solution.java
import java.util.*;

class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        int[] rank = {6,6,5,4,3,2,1};
        int max = 0;
        int min = 0;
        
        Arrays.sort(lottos);
        Arrays.sort(win_nums);
        
        for (int i=0; i<lottos.length; i++) {
            if (lottos[i] == 0) max++;
            if (Arrays.binarySearch(win_nums, lottos[i]) > -1) {
                max++;
                min++;
            }
        }
                
        answer[0] = rank[max];
        answer[1] = rank[min];
        
        return answer;
    }
}